templates/.circleci/config.yml

```diff
@@ -0,0 +1,237 @@
+
+      - save_cache: # Store cache
+          key: v1-pkg-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - "node_modules"
+
+      - store_test_results:
+          path: test-reports
+
+      - store_artifacts:
+          path: test-reports
+
+  build:
+    docker:
+      - image: *build-image
+    working_directory: *workspace
+    parameters:
+      config:
+        description: |
+          The config file you want to build the app using.
+          E.g. 'development' would load configurations from 'config/development.json'
+        type: enum
+        enum: ["development", "staging", "production"]
+        default: production
+    steps: # steps that comprise the `build` job
+      - attach_workspace:
+          at: *workspace
+      - restore_cache: # restores saved cache if no changes are detected since last run
+          keys:
+            - v1-pkg-cache-{{ checksum "package-lock.json" }}
+            - v1-pkg-cache-
+
+      - run:
+          name: Build static side
+          command: |
+            npm i
+            GATSBY_ACTIVE_ENV=<< parameters.config >> npm run build
+            mv public build_<< parameters.config >>
+
+      - persist_to_workspace:
+          root: /home/circleci/project
+          paths:
+            - build_<< parameters.config >>
+
+      - save_cache: # Store cache
+          key: v1-pkg-cache-{{ checksum "package-lock.json" }}
+          paths:
+            - "node_modules"
+
+      # Requires the SLACK_WEBHOOK env var to be set
+      # - slack/notify-on-failure
+
+  deploy:
+    docker:
+      - image: cimg/python:3.6
+    parameters:
+      config:
+        description: |
+          The config file you want to use to build your gatsby site.
+          E.g. 'development' would load configurations from '.env.development'
+
+          Note: .env.* file should not be added checked into source control rather should be
+          through your CircleCi's build environtemt variables see: https://circleci.com/docs/2.0/env-vars/#environment-variable-usage-options
+        type: enum
+        enum: ["development", "staging", "production"]
+        default: production
+      bucket:
+        description: The S3 Bucket to sync files to.
+        type: string
+      region:
+        type: string
+        default: ''
+        description: |
+          The region to use for AWS operations.
+    steps:
+      - attach_workspace:
+          at: *workspace
+      - run: # Required by aws-cli
+          name: Set AWS region var
+          command: echo 'export AWS_REGION=<< parameters.region >>' >> $BASH_ENV
+      - aws-s3/sync:
+          from: build_<< parameters.config >>
+          overwrite: true
+          to: 's3://<< parameters.bucket >>'
+      - run:
+          name: Invalidate Cloudfront
+          command: |
+            export DIST_ID=$(aws cloudfront list-distributions --query "DistributionList.Items[?Aliases.Items[?contains(@, '<< parameters.bucket >>')]].Id | [0]" | tr -d '"')
```

Comment:
If the bucket name is the domain name, and this will be using the root domain, won't `contains` match too many things? For example, I think there would probably be buckets called `domain.com` and `app.domain.com`, and this statement would match them both and potentially invalidate the wrong distribution.

Reply:
Right, good catch! I lifted that line from the frontend CI pipeline config -  both would need to change to be more specific.